src/core/src/tensor_conversion_util.cpp

```diff
@@ -18,7 +18,7 @@ Tensor wrap_tensor(const ngraph::HostTensorPtr& t) {
     } else if (p_shape.is_static()) {
         return {et, p_shape.to_shape(), t->get_data_ptr()};
     } else {
-        return {et, make_dynamic_shape()};
+        return {et, ov::Shape(p_shape.rank().get_length())};
```

Comment:
Why is this changed?
It provides correct conversion between dynamic host tensor and ov::tensor in evaluates and  this WA fix a lot of issues until old tensor will be removed.


@ilyachur can you have a look?

Reply:
Obviously this changes do not work. Need to discuss working with `ov::Tensor`  in the case of dynamics shapes.
The main problem is that `set_shape` call to the `ov::Tensor` instance created from "dynamic" `ov::Shape`, which can not really represent a dynamic shape and that is why we use `ov::PartialShape`, can not be performed since in that case the tensor object is in the "uninitialized" state. But, what should the `evaluate` implementation do in that case? That is the question. Because following the interface of the `ov::Tensor` it seems that this is enough to simply call `set_shape`, but it turns out that it throws if the tensor is not initialized. In other words, `ov::Tensor` is not designed to have "undefined" state, but we need this to at least express operations with "internal" dynamism.

Reply:
There is assumption that if `ov::Tensor` is create as "dynamic", `set_shape` will allow change to any shape. This is already used in [softsign evaluate](https://github.com/openvinotoolkit/openvino/blob/27ea9eab329cd7db547f5aeb24bfca7e549ef489/src/core/src/op/softsign.cpp#L95), and there is a case when this evaluate is called with "dynamic" output tensor and reshaped.

Maybe `set_shape` WA should be improved?

Reply:
It looks like I mixed up `ov::Tensor` and `runtime::Tensor`, since the letter being initialized from "dynamic" `ov::Shape` can not perform `set_shape` call, because an exception is thrown. But, `ov::Tensor` performs `set_shape` properly, even though being instantiated from the result of the `ov::util::make_dynamic_shape()` call.  So I adopted the `ov::util::make_dynamic_shape()` utility on the plugin level.